JKNS-630: Support Matrix for OpenShift Jenkins releases (#91968)
openshift/openshift-docs: modules/relocation-of-openshift-jenkins-images.adoc

Reviewer: There are several instances of `OpenShift` in plain text. Typically, these should all be replaced by `{product-title}`. I notice some of these instances are matching the use in the linked KB article, so I guess it's better to be consistent. I have suggested replacements where it's not in direct conflict with the article.

Related question: is "OpenShift Jenkins" a product name?

Author: Yes, "OpenShift Jenkins" is a product name. See [product page](https://pp.engineering.redhat.com/ocp_tools/overview/).

----
$ oc import-image jenkins-agent-maven -n openshift
----

> .What is the Jenkins support matrix for OpenShift, including supported versions and architectures?

Reviewer: This seems to answer a different question than is written, maybe:
```suggestion
.What {product-title} architectures and versions does Jenkins support?
```

Jenkins supports the following architectures across {product-title} releases.

> - amd64

Reviewer 2: ```suggestion
- `amd64`
```

> - arm64

Reviewer 2: ```suggestion
- `arm64`
```

> - ppc64le

Reviewer 2: ```suggestion
- `ppc64le`
```

> - s390x

Reviewer 2: ```suggestion
- `s390x`
```

> However, for OpenShift Extended Update Support (EUS) releases, only the amd64 architecture is officially supported. As a result, OpenShift Jenkins images are shipped exclusively for amd64 on these releases. This is because the {product-title} platform itself supports only the amd64 architecture for EUS releases. Refer to link:https://access.redhat.com/articles/7115356[OpenShift Jenkins Support Matrix] for more information.

Reviewer 2: ```suggestion
However, for OpenShift Extended Update Support (EUS) releases, only the `amd64` architecture is officially supported. As a result, OpenShift Jenkins images are shipped exclusively for `amd64` on these releases. This is because the {product-title} platform itself supports only the `amd64` architecture for EUS releases. Refer to link:https://access.redhat.com/articles/7115356[OpenShift Jenkins Support Matrix] for more information.
```
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference  I think these need to be in backticks based on this guide.

> However, for OpenShift Extended Update Support (EUS) releases, only the `amd64` architecture is officially supported. As a result, OpenShift Jenkins images are shipped exclusively for `amd64` on these releases. This is because the {product-title} platform itself supports only the `amd64` architecture for EUS releases. Refer to link:https://access.redhat.com/articles/7115356[OpenShift Jenkins Support Matrix] for more information.

Reviewer: Suggest either using the attribute:
```suggestion
However, for {product-title} Extended Update Support (EUS) releases, only the `amd64` architecture is officially supported. As a result, OpenShift Jenkins images are shipped exclusively for `amd64` on these releases. This is because the {product-title} platform itself supports only the `amd64` architecture for EUS releases. For more information, see link:https://access.redhat.com/articles/7115356[Support Matrix for OpenShift Jenkins releases].
```
or removing the name:
```suggestion
However, for Extended Update Support (EUS) releases, only the `amd64` architecture is officially supported. As a result, OpenShift Jenkins images are shipped exclusively for `amd64` on these releases. This is because the {product-title} platform itself supports only the `amd64` architecture for EUS releases. For more information, see link:https://access.redhat.com/articles/7115356[Support Matrix for OpenShift Jenkins releases].
```

> .When Red Hat updates Jenkins container images, are they available for all OpenShift versions simultaneously?

Reviewer: ```suggestion
.When Red{nbsp}Hat updates Jenkins container images, are they available for all {product-title} versions simultaneously?
```

Yes, Jenkins container images are updated on a quarterly basis, and the updates are made available for all supported Jenkins images across all supported {product-title} releases.

.How long are the released Jenkins images supported?

> Red Hat supports only the latest Long-Term Support (LTS) version of the Jenkins core, as provided in our latest container images. We do not support multiple core versions. Our policy is to align with the latest Jenkins LTS version released by the upstream community.

Reviewer: ```suggestion
Red{nbsp}Hat supports only the latest Long-Term Support (LTS) version of the Jenkins core, as provided in our latest container images. We do not support multiple core versions. Our policy is to align with the latest Jenkins LTS version released by the upstream community.
```

> .Does the Jenkins release align with OpenShift versions?

Reviewer: ```suggestion
.Does the Jenkins release align with {product-title} versions?
```

> Yes. Our goal is to maintain platform alignment—meaning Jenkins controller and agent images are built and tested for each supported {product-title} releases.

Reviewer: ```suggestion
Yes. Our goal is to maintain platform alignment. This means that Jenkins controller and agent images are built and tested for each supported {product-title} releases.
```

> .Are Jenkins release timelines aligned with OpenShift release cycles?

Reviewer: ```suggestion
.Are Jenkins release timelines aligned with {product-title} release cycles?
```

Jenkins is no longer part of the {product-title} core payload. Releases are managed separately. However, our intent is to publish updated OpenShift Jenkins images for newly released {product-title} releases within a few weeks of the {product-title} GA release.

> .Does Red Hat follow the upstream Jenkins lifecycle and LTS versions?

Reviewer: ```suggestion
.Does Red{nbsp}Hat follow the upstream Jenkins lifecycle and LTS versions?
```

> Yes. We align with the Jenkins upstream lifecycle and follow the LTS version. Red Hat typically ships OpenShift Jenkins image updates quarterly unless a critical fix requires an out-of-cycle release.

Reviewer: ```suggestion
Yes. We align with the Jenkins upstream lifecycle and follow the LTS version. Red{nbsp}Hat typically ships OpenShift Jenkins image updates quarterly unless a critical fix requires an out-of-cycle release.
```

To verify the current Jenkins LTS version:

> - Navigate to the Jenkins Catalog → Packages section

Reviewer: ```suggestion
- Navigate to the Jenkins Catalog -> Packages section
```

- Search for “Jenkins”

> - The result will show two packages—one of which is the Jenkins LTS package.

Reviewer: ```suggestion
- The result will show two packages, one of which is the Jenkins LTS package.
```
per https://www.ibm.com/docs/en/ibm-style?topic=punctuation-dashes#em-dashes

.What is not supported?
- Jenkins versions older than the current OpenShift Jenkins LTS are not supported.
- Running Jenkins outside of {product-title} is not supported.

> - Multiple core versions of Jenkins are not supported. Plugins bundled with our OpenShif Jenkins images follow the same versioning across all supported {product-title} releases.

Reviewer 3: nitpick, there is a typo `Openshif`

Author: Made the correction. Thanks!
